src/generic/GenericTypes.jl

```diff
@@ -798,6 +798,42 @@ end
 
 const PuiseuxSeriesElem{T} = Union{PuiseuxSeriesRingElem{T}, PuiseuxSeriesFieldElem{T}} where T <: RingElement
 
+###############################################################################
+#
+#   AbsMSeriesRing / AbsMSeries
+#
+###############################################################################
+
+mutable struct AbsMSeriesRing{T <: RingElement} <: AbstractAlgebra.MSeriesRing{T}
+   base_ring::Ring
+   poly_ring::MPolyRing{T}
+   prec_max::Vector{Int}
+   sym::Vector{Symbol}
+
+   function AbsMSeriesRing{T}(R::Ring, poly_ring::MPolyRing{T}, prec::Vector{Int}, s::Vector{Symbol}, cached::Bool = true) where T <: RingElement
+      if cached && haskey(AbsMSeriesID, (R, prec, s))
```

Comment:
Could you adjust this to be like the other caching things? Makes it more easy to enable the WeakValueDict once we get it working.

Reply:
Done.